plugins/events/create.js

```diff
@@ -19,270 +20,216 @@ module.exports = () => ({
-                                                        parentEvent.meta &&
-                                                        parentEvent.meta.parameters
-                                                    ) {
-                                                        payload.meta.parameters = parentEvent.meta.parameters;
-                                                    }
-
-                                                    if (!prNum) {
-                                                        payload.workflowGraph = parentEvent.workflowGraph;
-                                                        payload.sha = parentEvent.sha;
-
-                                                        if (parentEvent.configPipelineSha) {
-                                                            payload.configPipelineSha = parentEvent.configPipelineSha;
-                                                        }
-                                                    }
-                                                });
-                                            }
-
-                                            return Promise.resolve();
-                                        })
-                                        .then(() => eventFactory.create(payload))
-                                );
-                            })
-                            .then(event => {
-                                if (event.builds === null) {
-                                    return boom.notFound('No jobs to start.');
-                                }
-                                // everything succeeded, inform the user
-                                const location = urlLib.format({
-                                    host: request.headers.host,
-                                    port: request.headers.port,
-                                    protocol: request.server.info.protocol,
-                                    pathname: `${request.path}/${event.id}`
-                                });
-
-                                return h
-                                    .response(event.toJson())
-                                    .header('Location', location)
-                                    .code(201);
-                            });
-                    }
-                )
-                .catch(err => {
-                    throw err;
-                });
+                }
+            }
+
+            const event = await eventFactory.create(payload);
+
+            if (event.builds === null) {
+                return boom.notFound('No jobs to start.');
+            }
+            // everything succeeded, inform the user
+            const location = urlLib.format({
+                host: request.headers.host,
+                port: request.headers.port,
+                protocol: request.server.info.protocol,
+                pathname: `${request.path}/${event.id}`
+            });
+
```

Comment:
Refactoring into smaller functions would be nice and make it more readable. (can update later)

Reply:
I'll save that for a future PR